domainservice/src/main/kotlin/com/makarytskyi/rentcar/config/NatsListener.kt

```diff
@@ -0,0 +1,16 @@
+package com.makarytskyi.rentcar.config
+
+import io.nats.client.Connection
+import io.nats.client.ConnectionListener
+import org.springframework.stereotype.Component
+
+@Component
+class NatsListener : ConnectionListener {
```

Comment:
why do you need this class?

Reply:
To check status of connection to NATS, if it is disconnected, I stop consume from kafka topic items and wait until it is connected.